src/analysis/processing/qgsalgorithmmergevector.cpp

```diff
@@ -56,7 +56,8 @@ QString QgsMergeVectorAlgorithm::shortHelpString() const
 {
   return QObject::tr( "This algorithm combines multiple vector layers of the same geometry type into a single one.\n\n"
                       "If attributes tables are different, the attribute table of the resulting layer will contain the attributes "
-                      "from all input layers. New attributes will be added for the original layer name and source.\n\n"
+                      "from all input layers. If fields with the same name but different types are found, they are merged "
```

Comment:
leave the part about the new attributes too?

Reply:
I didn't understand what it referred to (I thought it was some leftover) until I tested the alg after your feedback; it adds a field storing the original layer name and another one with the path. It was not obvious to me when reading ;(
What about:
```
This algorithm combines multiple vector layers of the same geometry type into a single one.

The attribute table of the resulting layer will contain the fields from all input layers (*).
If fields with the same name but different types are found, they are merged into a string type field.
New fields storing the original layer name and source are also added.
```

(\*) should we add that identical fields are combined or is it implicit? Opinions? Thanks

Reply:
I like your wording. I don't think it's necessary to state that identicial fields are combined.

Just one tiny suggestion:

   If fields with the same name but different types are found, they are merged into a string type field.

maybe "If fields with the same name but different types are found then the exported field will be automatically converted into a string type field."